apollo-federation/src/sources/connect/validation/test_data/body_selection.graphql

```diff
@@ -61,4 +61,28 @@ type Query {
         }
         selection: "$"
     )
+    invalidArrowMethod(userid: ID!): String
+    @connect(
+        http: {
+            POST: "http://127.0.0.1",
+            body: "$({ userid: $args.userid })->no_such_method" # INVALID - no such method
+        }
+        selection: "$"
+    )
+    invalidVariable(userid: ID!): String
+    @connect(
+        http: {
+            POST: "http://127.0.0.1",
+            body: "$({ userid: $nosuchvariable })" # INVALID - no such variable
+        }
+        selection: "$"
+    )
+    stringLiteralFromArrowMethod(userid: ID!): String
+    @connect(
+        http: {
+            POST: "http://127.0.0.1",
+            body: "$({ userid: $args.userid })->jsonStringify" # INVALID - results in string literal
```

Comment:
this surprised me but i see the logic in it. we'll probably eventually support string values when you set `content-type: text/plain` or something, but let's assume for now that a non-object/non-array is a mistake. we should document this and probably list it as a breaking change in the preview changelog